### DISCUSSION TITLE
Solution: LockBufferForWrite: Multiple uploads in flight

### DISCUSSION THREAD

**Discussion by andymitchellx** [Antoshidza/NSprites#13] [General]
Solution: I switched from Direct3D11 to Direct3D12 (Project Settings => Player => Other Settings => Graphics APIs)

I think you worked this out before (https://github.com/Antoshidza/NSprites/discussions/8) but I was unfamiliar with the settings or cause.  So I thought I'd post this in case it helps others.

Symptoms before switch:

1. consistent Console log messages "multiple uploads in flight for buffer " when switching from one scene to another.
2. sprites consistently but randomly disappearing in the editor and during play
3. sprites disappearing in executable builds
4. occasionally, an executable would go "wonky" and start spewing random graphics, noise and triangles across the screen

After the switch:

1. no Console log messages
2. sprites are not disappearing
3. executable builds are stable

Research:

I tried lots of things over the past couple of days including a) upgrading and downgrading Unity, b) camera settings, c) post processing, d) debugging NSprites and e) verifying the BeginWrite/EndWrites in InstancedProperty.cs.  

Some references
https://github.com/sebas77/MillionsCubes/blob/main/Assets/DrawCubes.cs
https://forum.unity.com/threads/entity-rendering-and-compute-buffer-begin-write-gpu-synchronization.1371480/
https://forum.unity.com/threads/computebuffer-beginwrite.1252560/
https://forum.unity.com/threads/computebuffer-beginwrite-takes-a-long-time-if-program-like-obs-open-broadcaster-software-are-activ.943646/

**Comment by Antoshidza** [comment]
I guess this happens because different graphics API expect different sequence of data. This is why I've described API compatibility section in readme.md and in project's wiki.
I hope compatibility can be improved by rewriting synchronization logic using `BatchRendererGroup`. This is what Entities Graphics use to do the same thing that NSprites does.
